File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int n = word1.length();
        int m = word2.length();

        StringBuilder word = new StringBuilder();

        boolean temp = true;
        int i = 0;
        int j = 0;

        while (i < n && j < m) {
            if (temp) {
                word.append(word1.charAt(i));
                i++;
            } else {
                word.append(word2.charAt(j));
                j++;
            }

            temp = !temp;
        }

        while (i < n) {
            word.append(word1.charAt(i));
            i++;
        }

        while (j < m) {
            word.append(word2.charAt(j));
            j++;
        }

        return word.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();
        string word;
        bool temp = true;
        int i = 0;
        int j = 0;
        while(i<n && j<m){
            if(temp){
                word.push_back(word1[i]);
                i++;
            }else{
                word.push_back(word2[j]);
                j++;
            }
            temp = !temp;
        }

        while(i<n){
            word.push_back(word1[i]);
            i++;
        }
        while(j<m){
            word.push_back(word2[j]);
            j++;
        }

        return word;
    }
};